core/contracts/oracle/implementation/ContractCreator.sol

```diff
@@ -8,14 +9,14 @@ import "./Constants.sol";
 /**
  * @title Base contract for all financial contract creators
  */
-abstract contract ContractCreator {
+abstract contract ContractCreator is ReentrancyGuard {
     address internal finderAddress;
 
     constructor(address _finderAddress) public {
         finderAddress = _finderAddress;
     }
 
-    function _registerContract(address[] memory parties, address contractToRegister) internal {
+    function _registerContract(address[] memory parties, address contractToRegister) internal nonReentrant() {
```

Comment:
Added because of call to `registry.registerContract`

Reply:
So just to double-check: reentrancy would occur here if the party registered was a smart contract with a hook it could re-enter at this point? what would this enable it to do within the context of the `_registerContract` function?

Reply:
I will remove though because its not an external method